smartsim/experiment.py

```diff
@@ -143,10 +158,9 @@ def __init__(
 
         :param name: name for the ``Experiment``
         :param exp_path: path to location of ``Experiment`` directory
-        :param launcher: type of launcher being used, options are "slurm", "pbs",
-                         "lsf", "sge", or "local". If set to "auto",
-                         an attempt will be made to find an available launcher
-                         on the system.
+        :param settings_dispatcher: The dispatcher the experiment will use to
+            figure determine how to launch a job. If none is provided, the
+            experiment will use the default dispatcher.
         """
         self.name = name
         if exp_path:
```

Comment:
> Since we are importing all of os should we use os.path? Or do we not need ot be importing all of os?

At the moment are still using `os` for `os.makedirs` in `Experiment._create_entity_dir` so we cannot get rid of `import os`. The `import os.path as osp` at that point is really just acting as an alias for the the namespace. I have no strong preference as long as we are consistent throughout the file. I'll update everything to `osp`!

In the long term though (outside of this PR) I think it makes sense to promote `Experiment.exp_path` to a `pathlib.Path` and remove most (all?) of the use of `os` within this module.

>  Is that still the behavior we want?

I am 100% in favor of changing this behavior, but I'd rather bump it to its own "short task" rather than sneak the change in with this PR

Reply:
Everything has been moved over to use `osp` where applicable!

Reply:
new ticket to be made